--- a/libntech/configure.ac.old	2022-11-28 09:46:05.826563092 +0100
+++ b/libntech/configure.ac	2022-11-28 09:46:18.677883295 +0100
@@ -573,23 +573,6 @@
     fi
 fi
 
-CF3_WITH_LIBRARY(libxml2,
-    [AC_CHECK_LIB(xml2, xmlFirstElementChild,
-    [],
-    [if test "x$with_libxml2" != xcheck; then
-        AC_MSG_ERROR(Cannot find libxml2); fi]
-    )
-    AC_CHECK_HEADERS([libxml/xmlwriter.h], [break],
-        [if test "x$with_libxml2" != xcheck; then
-            AC_MSG_ERROR(Cannot find libxml2); fi]
-    )]
-)
-
-AM_CONDITIONAL([HAVE_LIBXML2],
-    [test "x$with_libxml2" != xno &&
-     test "x$ac_cv_lib_xml2_xmlFirstElementChild" = xyes])
-
-
 dnl ######################################################################
 dnl Checks for header files.
 dnl ######################################################################
