--- a/misc/CMakeLists.txt
+++ b/misc/CMakeLists.txt
@@ -49,6 +49,7 @@ if (PCAP_CONFIG)
   execute_process (COMMAND pcap-config --libs
     OUTPUT_VARIABLE PCAP_LDFLAGS
     OUTPUT_STRIP_TRAILING_WHITESPACE)
+  string(STRIP ${PCAP_LDFLAGS} PCAP_LDFLAGS)
   execute_process (COMMAND pcap-config --cflags
     OUTPUT_VARIABLE PCAP_CFLAGS
     OUTPUT_STRIP_TRAILING_WHITESPACE)
--- a/nasl/CMakeLists.txt
+++ b/nasl/CMakeLists.txt
@@ -69,6 +69,7 @@ if (PCAP_CONFIG)
   execute_process (COMMAND pcap-config --libs
     OUTPUT_VARIABLE PCAP_LDFLAGS
     OUTPUT_STRIP_TRAILING_WHITESPACE)
+  string(STRIP ${PCAP_LDFLAGS} PCAP_LDFLAGS)
   execute_process (COMMAND pcap-config --cflags
     OUTPUT_VARIABLE PCAP_CFLAGS
     OUTPUT_STRIP_TRAILING_WHITESPACE)
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -69,6 +69,7 @@ if (PCAP_CONFIG)
   execute_process (COMMAND pcap-config --libs
     OUTPUT_VARIABLE PCAP_LDFLAGS
     OUTPUT_STRIP_TRAILING_WHITESPACE)
+  string(STRIP ${PCAP_LDFLAGS} PCAP_LDFLAGS)
   execute_process (COMMAND pcap-config --cflags
     OUTPUT_VARIABLE PCAP_CFLAGS
     OUTPUT_STRIP_TRAILING_WHITESPACE)
