diff --git a/speed-test/cryptoperf-skcipher.c b/speed-test/cryptoperf-skcipher.c
index 9dce932..a3f257f 100644
--- a/speed-test/cryptoperf-skcipher.c
+++ b/speed-test/cryptoperf-skcipher.c
@@ -159,6 +159,12 @@ static const struct cp_skcipher_tests testcases[] = {
 	{ "AES(G) CBC(G) 128", "cbc(aes-generic)", 16 },
 	{ "AES(G) CBC(G) 192", "cbc(aes-generic)", 24 },
 	{ "AES(G) CBC(G) 256", "cbc(aes-generic)", 32 },
+	{ "AES(CE) CBC(CE) 128", "cbc-aes-sun8i-ce", 16 },
+	{ "AES(CE) CBC(CE) 192", "cbc-aes-sun8i-ce", 24 },
+	{ "AES(CE) CBC(CE) 256", "cbc-aes-sun8i-ce", 32 },
+	{ "AES(CE) CBC(CE) 128", "cbc-aes-sun4i-ss", 16 },
+	{ "AES(CE) CBC(CE) 192", "cbc-aes-sun4i-ss", 24 },
+	{ "AES(CE) CBC(CE) 256", "cbc-aes-sun4i-ss", 32 },
 	{ "AES(AESNI) CBC(ASM) 128", "cbc-aes-aesni", 16 },
 	{ "AES(AESNI) CBC(ASM) 192", "cbc-aes-aesni", 24 },
 	{ "AES(AESNI) CBC(ASM) 256", "cbc-aes-aesni", 32 },
--- a/speed-test/cryptoperf-hash.c.old	2017-05-28 13:04:27.000000000 +0200
+++ b/speed-test/cryptoperf-hash.c	2017-05-28 13:05:31.000000000 +0200
@@ -132,6 +132,8 @@
 	{ "SHA-256(G)", "sha256-generic", 0 },
 	{ "SHA-384(G)", "sha384-generic", 0 },
 	{ "SHA-512(G)", "sha512-generic", 0 },
+	{ "MD5(SS)", "md5-sun4i-ss", 0 },
+	{ "SHA-1(SS)", "sha1-sun4i-ss", 0 },
 	{ "SHA-1(SSSE3)", "sha1-ssse3", 0 },
 	{ "SHA-224(SSSE3)", "sha224-ssse3", 0 },
 	{ "SHA-256(SSSE3)", "sha256-ssse3", 0 },

